### ISSUE TITLE
Got issues while making for iOS

### ISSUE THREAD

**Issue by kentkrantz** [Lua-cURL/Lua-cURLv3#141]
I got this error while building for iOS, but can't fix it.

```command
xcrun -sdk iphoneos gcc -arch armv7 -arch armv7s -arch arm64 -miphoneos-version-min=6.0 -O2 -g -pipe -fPIC -ggdb -DPTHREADS -std=c99 -I../../../include -I../../../include/lua -c -o l52util.o l52util.c
xcrun -sdk iphoneos gcc -arch armv7 -arch armv7s -arch arm64 -miphoneos-version-min=6.0 -O2 -g -pipe -fPIC -ggdb -DPTHREADS -std=c99 -I../../../include -I../../../include/lua -c -o lceasy.o lceasy.c
In file included from lceasy.c:19:
./lcurlapi.h:26:2: error: expected ';' after struct
}lcurl_url_t;
 ^
 ;
./lcurlapi.h:22:1: warning: typedef requires a name [-Wmissing-declarations]
typedef struct lcurl_url_tag {
^~~~~~~
1 warning and 1 error generated.
make: *** [lceasy.o] Error 1
```

**Comment by moteus** [comment]
There exists PR about this i think.
I'll try to checkout it tomorrow.

**Comment by moteus** [comment]
Can you please test fix from #142?
I am not sure how to reproduce this error on my test environment, but I think I understand what is going on.

**Comment by kentkrantz** [comment]
@moteus It's fixed , you are great!
Got these warnings:
```shell
lcmime.c:186:20: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
lcmime.c:186:20: note: place parentheses around the assignment to silence this warning
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
                   ^
            (                                 )
lcmime.c:186:20: note: use '==' to turn this assignment into an equality comparison
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
                   ^
                   ==
1 warning generated.
lcmime.c:186:20: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
lcmime.c:186:20: note: place parentheses around the assignment to silence this warning
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
                   ^
            (                                 )
lcmime.c:186:20: note: use '==' to turn this assignment into an equality comparison
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
                   ^
                   ==
1 warning generated.
lcmime.c:186:20: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
lcmime.c:186:20: note: place parentheses around the assignment to silence this warning
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
                   ^
            (                                 )
lcmime.c:186:20: note: use '==' to turn this assignment into an equality comparison
    for(i=0;method = lcurl_mime_part_fields[i]; ++i){
                   ^
                   ==
1 warning generated.

```

**Comment by moteus** [comment]
Closed by #142 
About this warning. You can open new issue or PR
